### ISSUE TITLE
For-of-loop in test

### ISSUE THREAD

**Issue by wassy92x** [Typescript-TDD/ts-auto-mock#210]
Unfortunately I found another bug.
As soon as I create a mock of an interface, I can't use For-of-loops in my tests.
I always get an exception: "TypeError: Cannot read property 'ɵRepository' of undefined"
If I comment out createMock, the for-of-loop works correctly.

Example:
```
import { createMock } from 'ts-auto-mock';

interface MyInterface {
    a: string;
}

describe('Foo', () => {
    test('some test', () => {
        const myInstance: MyInterface = createMock<MyInterface>();
        const a: string[] = ['hello', 'world'];
        for (const b of a) {
            console.log(b);
        }
    });
});
```

**Comment by Pmyl** [comment]
Could you tell me your configuration to run it? (Webpack, jasmine, ts-loader, etc)

It could be something related to the fact that for..of creates some polyfill emits that for some reason delete our imports.

I’ll look into it unless @uittorio wants to take this

**Comment by wassy92x** [comment]
My test environment is the following:
- jest-cli: 23.6.0
- jest: 23.6.0
- ts-jest: 23.10.5
- ttypescript: 1.5.10
- typescript: 3.7.3

Jest configuration:
```
{
   [...],
    "testMatch": [
      "**/test/**/*.test.js?(x)",
      "**/test/**/*.test.ts?(x)"
    ],
    "moduleDirectories": [
      "node_modules",
      "src/ts",
      "src/js"
    ],
    "moduleFileExtensions": [
      "js",
      "ts",
      "tsx",
      "json"
    ],
    "transform": {
      "^.+\\.tsx?$": "ts-jest",
      "^.+\\.jsx?$": "babel-jest"
    },
    "globals": {
      "ts-jest": {
        "compiler": "ttypescript"
      }
    }
}
```

tsconfig.json
{
  "compilerOptions": {
    "outDir": "./dist/",
    "noImplicitAny": true,
    "module": "es6",
    "target": "es5",
    "jsx": "react",
    "allowJs": true,
    "sourceMap": true,
    "lib": ["es7", "dom"],
    "downlevelIteration": true,
    "allowSyntheticDefaultImports": true,
    "experimentalDecorators": true,
    "moduleResolution": "node",
    "declaration": false,
    "emitDecoratorMetadata": true,
    "skipLibCheck": true,
    "esModuleInterop": true,
    "plugins": [
      {
        "transform": "ts-nameof",
        "type": "raw"
      },
      {
        "transform": "ts-auto-mock/transformer",
        "cacheBetweenTests": false
      }
    ]
  },
  "include": [
    "src/ts/**/*.ts",
    "src/ts/**/*.tsx",
    "src/ts/**/*.d.ts",
    "test/ts/**/*.ts",
    "test/ts/**/*.tsx"
  ],
  "exclude": [
    "node_modules"
  ]
}

**Comment by uittorio** [comment]
Wow! Really interesting issue :)

Thanks for reporting this! I can have a look thai evening and hopefully fix it

**Comment by wassy92x** [comment]
For your information: I don't use your jest-ts-auto-mock package.

**Comment by uittorio** [comment]
Hi @wassy92x I'm glad to tell you that we've found the issue. there is a name collision with variables. We are using a functionality for typescript that has a bug or we are using it in the wrong way.  I have a solution. I'll open the PR in few hours :)

**Comment by wassy92x** [comment]
I found out that the problem doesn't occure if you set the typescript-target to minimum ES6

**Comment by uittorio** [comment]
Thanks, I've just released a new version 1.5.1 of ts-auto-mock :)
